code/datums/weather/weather.dm

```diff
@@ -217,21 +294,42 @@
 /datum/weather/proc/update_areas()
 	for(var/V in impacted_areas)
 		var/area/N = V
+		if(stage == MAIN_STAGE && multiply_blend_on_main_stage)
+			N.blend_mode = BLEND_MULTIPLY
+		else
+			N.blend_mode = BLEND_OVERLAY
+		if(stage == MAIN_STAGE && opacity_in_main_stage)
+			N.set_opacity(TRUE)
+		else
+			N.set_opacity(FALSE)
 		N.layer = overlay_layer
 		N.plane = overlay_plane
 		N.icon = 'icons/effects/weather_effects.dmi'
 		N.color = weather_color
-		switch(stage)
-			if(STARTUP_STAGE)
-				N.icon_state = telegraph_overlay
-			if(MAIN_STAGE)
-				N.icon_state = weather_overlay
-			if(WIND_DOWN_STAGE)
-				N.icon_state = end_overlay
-			if(END_STAGE)
-				N.color = null
-				N.icon_state = ""
-				N.icon = 'icons/turf/areas.dmi'
-				N.layer = initial(N.layer)
-				N.plane = initial(N.plane)
-				N.set_opacity(FALSE)
+		set_area_icon_state(N)
+		if(stage == END_STAGE)
+			N.color = null
+			N.icon = 'icons/turf/areas.dmi'
+			N.layer = initial(N.layer)
+			N.plane = initial(N.plane)
+			N.set_opacity(FALSE)
+
+/datum/weather/proc/set_area_icon_state(area/Area)
+	switch(stage)
+		if(STARTUP_STAGE)
+			Area.icon_state = telegraph_overlay
+		if(MAIN_STAGE)
+			Area.icon_state = weather_overlay
+		if(WIND_DOWN_STAGE)
+			Area.icon_state = end_overlay
+		if(END_STAGE)
+			Area.icon_state = ""
+
+/datum/weather/proc/do_thunder()
+	var/picked_sound = THUNDER_SOUND
+	for(var/i in 1 to impacted_areas.len)
```

Comment:
```suggestion
	for(var/i in 1 to length(impacted_areas))
```

Reply:
The existance of `impacted_areas` is asserted. We can use .len